### PR TITLE
core: Add 243390-rpi3 support, use specific network configuration for it

### PR DESCRIPTION
**Pull request by acostach**
Change-type: patch
Signed-off-by: Alexandru Costache <alexandru@balena.io>

This PR also specifies that the DUT network connection file should be applied to the embedded wifi connection in the case of 243390-rp3, otherwise connectivity between the two devices fails after a DUT reboot: https://github.com/balena-os/leviathan/pull/770#issuecomment-1152351969

To fully support this DT in jenkins, the device contract will probably have to be included in a different way because it's not part of the public contracts.

### CODE REVIEW

**Comment by klutchell** [comment]
Can you split this in two commits? One for worker and one for core?

We are in the process of moving the worker to a separate repo as it's own product so we can cherry-pick your commit to a PR over there once it's ready.

**Comment by acostach** [comment]
Sure, thanks Kyle

**Comment by klutchell** [comment]
@acostach can you move the worker changes to a PR over here? https://github.com/balena-os/leviathan-worker

The core changes can stay in this PR.

**Comment by acostach** [comment]
Sure @klutchell, I moved that in https://github.com/balena-os/leviathan-worker/pull/7

**Review by vipulgupta2048 — APPROVED** [review]

**Comment by vipulgupta2048** [comment]
@balena-ci rebase

**Comment by acostach** [comment]
@balena-ci rebase

File: core/lib/components/os/balenaos.js [hunk 1]
```diff
@@ -94,6 +94,7 @@ const injectNetworkConfiguration = async (image, configuration, partition = 1) =
 	const wifiConfiguration = [
 		'[connection]',
 		'id=balena-wifi',
+		`${configuration.wireless.interfaceName}`,
```
[the remark this answers is not in the document]
Comment (acostach) [reply core/lib/components/os/balenaos.js:97]: @rcooke-warwick I don't think we can hardcode it for all devices, some of them use the new interface naming convention i.e wlp2s0, I think the NUC might be one of them. So we need to add it to the list of exceptions too if it does.

**Comment by rcooke-warwick** [comment]
hey @acostach I tried this fix with the 243390 DUT, and it still has wlan0 not connected:

```
root@c9f41ad:~# cat /mnt/boot/system-connections/balena-wifi  
[connection]
id=balena-wifi
interface-name=wlan0
type=wifi
[wifi]
hidden=true
mode=infrastructure
ssid=nka0qmal
[ipv4]
method=auto
[ipv6]
addr-gen-mode=stable-privacy
method=auto
[wifi-security]
auth-alg=open
key-mgmt=wpa-psk
root@c9f41ad:~# nmcli d
DEVICE     TYPE      STATE         CONNECTION         
eth0       ethernet  connected     Wired connection 1 
wlan1      wifi      connected     hotspot            
wlan0      wifi      disconnected  --                 
balena0    bridge    unmanaged     --                 
resin-dns  bridge    unmanaged     --                 
lo         loopback  unmanaged     --                 
root@c9f41ad:~# 
```

**Comment by rcooke-warwick** [comment]
Myabe the device specific test is leaving it in that state, is there something we have to change there?

**Comment by rcooke-warwick** [comment]
@acostach if a balena device has an ethernet connection already, will network manager to to use the wifi? Even when I don't do the device specific test first, the DUT is never `auto-activating connection 'balena-wifi'` like it does for device types with working wifi

**Comment by acostach** [comment]
@rcooke-warwick I suspect NM may not auto activate it because wlan1 connected first so there's already an active wifi connection. I need to test this to confirm

**Pull request merged**